tests/virtctl/memorydump.go

```diff
@@ -316,19 +322,34 @@ func verifyMemoryDumpFile(dumpFilePath, dumpName string) {
 	tarReader := tar.NewReader(gzReader)
 
 	for {
-		header, err := tarReader.Next()
+		var header *tar.Header
+		header, err = tarReader.Next()
 		if err == io.EOF {
 			break
 		}
 		Expect(err).ToNot(HaveOccurred())
 		switch header.Typeflag {
 		case tar.TypeDir:
-			Expect(os.MkdirAll(filepath.Join(extractPath, header.Name), 0750)).To(Succeed())
+			var path string
+			path, err = sanitizedPath(extractPath, header.Name)
+			Expect(err).ToNot(HaveOccurred())
+			const permRWX = 0o700
+			Expect(os.MkdirAll(path, permRWX)).To(Succeed())
 		case tar.TypeReg:
-			extractedFile, err := os.Create(filepath.Join(extractPath, header.Name))
+			var path string
+			path, err = sanitizedPath(extractPath, header.Name)
 			Expect(err).ToNot(HaveOccurred())
-			_, err = io.Copy(extractedFile, tarReader)
+			var extractedFile *os.File
+			extractedFile, err = os.Create(path)
 			Expect(err).ToNot(HaveOccurred())
+			for {
+				const megabyte = 1024 * 1024
+				_, err = io.CopyN(extractedFile, tarReader, megabyte)
+				if err != nil && err == io.EOF {
+					break
+				}
+				Expect(err).ToNot(HaveOccurred())
+			}
```

Comment:
Was this change triggered by the linter complaining about something? It is not trivial.

Reply:
Yes, it was something the linter complained about. I moved it to a separate commit.